### PR TITLE
[B2BP-194] - Action Workflow GitHub Build Deploy WebSite

### PR DESCRIPTION
**Pull request by maurodandrea**
Create Action and Workflow GitHub for Build and Deploy a static Web Site on bucket S3 AWS. Depends on #52 #58 

#### List of Changes
Script for Create Action and Workflow GitHub for Build and Deploy a static Web Site on bucket S3 AWS.

#### Motivation and Context
As reported in RFC, they are necessary script for create Action and Workflow GitHub for Build and Deploy a static Web Site on bucket S3 AWS.

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change not requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/deploy_website.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          environment: ${{ inputs.environment }}
+          path_to_gitbook_docs: ${{ github.workspace }}/apps/nextjs-website/docs
+          #cookie_domain_script: ${{ secrets.COOKIE_DOMAIN_SCRIPT }}
+          iam_role: ${{ secrets.DEPLOY_IAM_ROLE }}
+          bucket: ${{ vars.S3_BUCKET_NAME }}
+          cloudfront_distribution_id: ${{ vars.CLOUDFRONT_DISTRIBUTION_ID }}
+          #cognito_user_pool_id: ${{ secrets.NEXT_PUBLIC_COGNITO_USER_POOL_ID }}
+          #cognito_app_client_id: ${{ secrets.NEXT_PUBLIC_COGNITO_USER_POOL_WEB_CLIENT_ID }}
+          strapi_token: ${{ secrets.STRAPI_TOKEN }}
+          strapi_api_url: ${{ vars.STRAPI_URL }}
```
Comment (datalek) [inline .github/workflows/deploy_website.yaml:82]: Remove the commented variables

File: .github/workflows/deploy_website.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          environment: ${{ matrix.environment }}
+          path_to_gitbook_docs: ${{ github.workspace }}/apps/nextjs-website/docs
+          #cookie_domain_script: ${{ secrets.COOKIE_DOMAIN_SCRIPT }}
+          iam_role: ${{ secrets.DEPLOY_IAM_ROLE }}
+          bucket: ${{ vars.S3_BUCKET_NAME }}
+          cloudfront_distribution_id: ${{ vars.CLOUDFRONT_DISTRIBUTION_ID }}
+          #cognito_user_pool_id: ${{ secrets.NEXT_PUBLIC_COGNITO_USER_POOL_ID }}
+          #cognito_app_client_id: ${{ secrets.NEXT_PUBLIC_COGNITO_USER_POOL_WEB_CLIENT_ID }}
+          strapi_token: ${{ secrets.STRAPI_TOKEN }}
+          strapi_api_url: ${{ vars.STRAPI_URL }}
```
Comment (datalek) [inline .github/workflows/deploy_website.yaml:58]: Also here, remove the commented variables

File: .github/workflows/deploy_website.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+
+      - name: Deploy website
+        uses: ./.github/actions/deploy
+        with:
+          environment: ${{ matrix.environment }}
+          path_to_gitbook_docs: ${{ github.workspace }}/apps/nextjs-website/docs
```
Comment (datalek) [inline .github/workflows/deploy_website.yaml:50]: The system does not need the `path_to_gitbook_docs`

File: .github/workflows/deploy_website.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+
+      - name: Deploy website
+        uses: ./.github/actions/deploy
+        with:
+          environment: ${{ inputs.environment }}
+          path_to_gitbook_docs: ${{ github.workspace }}/apps/nextjs-website/docs
```
Comment (datalek) [inline .github/workflows/deploy_website.yaml:74]: Also here, the system does not need the `path_to_gitbook_docs`

File: .github/actions/deploy/action.yaml
Comment (datalek) [inline .github/actions/deploy/action.yaml:1]: Remove the useless variables (e.g.: All cognito related variables, and gitbook related variables)

**Review by datalek — CHANGES_REQUESTED** [review]

File: .github/actions/deploy/action.yaml
Comment (maurodandrea) [reply .github/actions/deploy/action.yaml:1]: ok, change made with the latest commit

File: .github/workflows/deploy_website.yaml (hunk 3 above)
Comment (maurodandrea) [reply .github/workflows/deploy_website.yaml]: ok, change made with the latest commit

File: .github/workflows/deploy_website.yaml (hunk 2 above)
Comment (maurodandrea) [reply .github/workflows/deploy_website.yaml]: ok, change made with the latest commit

File: .github/workflows/deploy_website.yaml (hunk 4 above)
Comment (maurodandrea) [reply .github/workflows/deploy_website.yaml]: ok, change made with the latest commit

File: .github/workflows/deploy_website.yaml (hunk 1 above)
Comment (maurodandrea) [reply .github/workflows/deploy_website.yaml]: ok, change made with the latest commit

File: .github/workflows/deploy_website.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+
+      - name: Deploy website
+        uses: ./.github/actions/deploy
+        with:
+          environment: ${{ inputs.environment }}
+          iam_role: ${{ secrets.DEPLOY_IAM_ROLE }}
```
Comment (datalek) [inline .github/workflows/deploy_website.yaml:70]: Can you create a PR to create the IAM ROLE that we should use here? It should have limited permission, you can check [this as example](https://github.com/pagopa/developer-portal/blob/main/.infrastructure/40_iam.tf#L3-L78)

Comment (maurodandrea) [reply .github/workflows/deploy_website.yaml:70]: Yes, now I create branches and PRs for role creation

Comment (maurodandrea) [reply .github/workflows/deploy_website.yaml]: ok, i have open a PR https://github.com/pagopa/b2b-portals/pull/58 for create new role:

1) GitHubActionDeployECS for deploy CMS on ECS. In Workflow i have write secrets.IAM_ROLE_DEPLOY_CMS
2) GitHubActionDeployWebsite for deploy a static website on S3. In Workflow i have write secrets.IAM_ROLE_DEPLOY_WEBSITE

File: .github/workflows/deploy_website.yaml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        type: choice
+        required: true
+        default: prod
+        options:
+          - prod
+
+  # Allows external webhook trigger
+  repository_dispatch:
+    types:
+      - webhook
```
[the remark this answers is not in the document]
Comment (maurodandrea) [reply .github/workflows/deploy_website.yaml:22]: I have added the condition github.event_name == 'repository_dispatch' inside the cd_deploy job

**Review by datalek — APPROVED** [review]
LGTM

**Pull request merged**